### ISSUE TITLE
Visual Options

### ISSUE THREAD

**Issue by reezlock** [casualshammy/NameplateAuras#67]
Love the functionality and the customization setup! 

The only things I'm missing are a couple of (mostly) visual options, maybe they could make it into the addon one day…

- [x] Is it possible to add a Pulse/Flash effect as an alternative/addition to the Glow effects when a Debuff is close to expiring?
- [x] I currently need specify the time until expiration for each Debuff as "seconds remaining". Would it be possible to specify "percent of max duration remaining" instead (i.e. 30% for Pandemic Windows)?
- [x] ~Adding Masque support/functionality would be a dream!~

**Comment by casualshammy** [comment]
Hello!
Thanks for suggestions. I think I could implement Masque support before next release (on 09/30). Other ideas will be implemented and tested in beta versions on October.

**Comment by reezlock** [comment]
Hi,

wonderful news, thanks for considering everything!

**Comment by casualshammy** [comment]
There is a problem. I don't know why, but Masque doesn't allow to decorate simple plain `Frame`, it requires that all icons to be `buttons`. It's a pain in ass because buttons create huge performance impact when they are moved around the screen like nameplates do.
Currenty I don't know how to negate fps drop, so `Masque` part is also delayed.

**Comment by reezlock** [comment]
That's really strange, seeing as you can skin Weakauras with Masque which aren't buttons at all (unless Icons in WA are also coded like this)…

**Comment by casualshammy** [comment]
All icons in WeakAuras have disabled buttons attached. This buttons are created only if Masque is installed and enabled.

<details> 
  <summary>Lua code</summary>

   ```lua
local MSQ, MSQ_Version = LibStub("Masque", true);
local button
if MSQ then
    button = CreateFrame("Button", nil, region)
    button.data = data
    region.button = button;
    button:EnableMouse(false);
    button:Disable();
    button:SetAllPoints();
end
```
</details>

**Comment by casualshammy** [comment]
There is a problem with
> (2) I currently need specify the time until expiration for each Debuff as "seconds remaining". Would it be possible to specify "percent of max duration remaining" instead (i.e. 30% for Pandemic Windows)?

NameplateAuras doesn't know anything about aura's maximum duration itself, it knows only *current* maximum duration. For example, you have some DoT spell that lasts 10 seconds. Okay, NameplateAuras will pulse/glow icon when this DoT's remaining time is 3 sec. Then, you re-apply this DoT. Maximum duration is 13 sec now. This time NameplateAuras will pulse/glow icon on 4 seconds remaining (13*30%=3.9). Is it OK?

**Comment by reezlock** [comment]
That's exactly as it should be!

The pandemic window is always relative to the current maximum duration. That's kept in check by the fact that no matter how often and how early you refresh, the dot from your example will never be longer than those 13 secs. Thus, every pandemic window will be between 30% and 39% of the base maximum duration, but never more.

**Comment by casualshammy** [comment]
- [x] Pulse/Flash effect
- [ ] Pandemic Windows
- [ ] Masque support

I will appreciate if you'll find spare time to try out the latest beta (you can install it via Twitch client or download [directly](https://www.curseforge.com/wow/addons/nameplateauras/download/3070293)).

**Comment by reezlock** [comment]
I played around with the two new effects a little bit.

The Flash effect is perfect as it is.

The Pulse effect seems to be anchored at the top left of the icon, so that it grows to the botttom right. This should probably be centered.

Also, the size of the Pulse effect is pretty large for me, but that is probably because my icons are relatively small (17 or 18 px).

I guess if you do not want to add sliders for Flash frequency and Pulse size (luxury and all that), you could tie the Pulse effect to an icon scale parameter, where you just set a fixed pulse size that looks right for, say, 40px icons and then automatically scale that proportionally to where the user sets the icon size. (i.e.: if 20px icon, then 50% pulse size etc.)

**Comment by casualshammy** [comment]
TBH I just can't make scale animation work :D So now it's just a stub. I'll return to it a bit later.
Also, maybe I'll make flash animation progressive (less time remains - more frequent). I think it's quite good solution.
Masque support will not be implemented. I have managed to negate impact on fps , but... Um, it is not just 'call function from Masque and style is applied', it requires to add 'another way of doing things' to spell handling mechanism just because Masque supports only buttons, but NAuras icons are _not_ buttons and was _not_ intended to work as buttons from the very begining of developing this addon.

**Comment by casualshammy** [comment]
Pandemic Windows.
I will appreciate if you'll find spare time to try out the latest beta (you can install it via Twitch client or download [directly](https://www.curseforge.com/wow/addons/nameplateauras/download/3074738)). WOW client restart is required.

**Comment by reezlock** [comment]
Tested the new features, both the custom border texture option and the relative time (pandemic) support are working perfectly!

I am seeing some bugs with the timer text color/alphas though:

(1) The alpha slider seems to be inverted, ie. 0 alpha at the top ("+") of the slider and 100 alpha at the bottom ("-").
(2) The defined colours for each of the time segments (>1m, <1m, <5s) work fine when applying an fresh debuff to a target, but once they are refreshed while still running, they change their behaviour, sometimes just inverting the alpha value and sometimes changing the alpha of other segments.

Example: I have set the color of the <1m segment to "white, 50% alpha" and the <5s segment to "red, 100% alpha". When I refresh the debuff at <5s, it becomes "white, 100% alpha".

Example 2: I set color of the <1m segment to "white, 0% alpha" and the <5s segment to "red, 100% alpha". When I refresh the debuff at <5s, it becomes "white, 100% alpha" AND when it reaches <5, it becomes "red(?), 0% alpha.
